### ISSUE TITLE
dynamic event loading

### ISSUE THREAD

**Issue by scott2b** [NUKnightLab/TimelineJS3#418]
Now that timelines can be backed by json, a next-level feature would be to allow for dynamic loading of timeline events.

See, for example, [this zendesk ticket](https://knightlab.zendesk.com/agent/tickets/3506)

As in that ticket, one issue to consider is appropriate timeline event layout when loading dynamically. The cited user handled this problem by loading all events before generating the timeline.

**Comment by zachwise** [comment]
The media etc is not loaded until it's a couple slides away, so it already dynamically loads content. Is this what you're talking about.

**Comment by scott2b** [comment]
I don't think so. In this case, the user was programmatically loading event data. I would describe what you are referring to as either lazy-loading, or lazy-rendering depending on what exactly the code is doing.
